pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -420,11 +427,26 @@ private void cleanCacheAndCloseReader(@Nonnull NamespaceName namespace, boolean
         });
     }
 
+    private void completeResults(TopicName topicName) {
+        if (results.get(topicName) != null) {
+            BlockingDeque<CompletableFuture> futures = results.get(topicName);
+            CompletableFuture<Void> future;
+            while (true) {
+                future = futures.poll();
+                if (future == null) {
+                    break;
+                }
+                future.complete(null);
+            }
+        }
+    }
+
     private void readMorePolicies(SystemTopicClient.Reader<PulsarEvent> reader) {
         reader.readNextAsync()
                 .thenAccept(msg -> {
                     refreshTopicPoliciesCache(msg);
                     notifyListener(msg);
+                    completeResults(TopicName.get(TopicName.get(msg.getKey()).getPartitionedTopicName()));
```

Comment:
I dont think this is a right way to solve the problem. For examle"

If we have three different actions, e.g., `setA` and `setB` and `setC`:
1. setA is waiting for  futureA completed. The eventMsg is eventA.
2. setB  is waiting for futureB completed. The eventMsg is eventB.
3. setC is waiting for futureC completed. The eventMsg is eventC.
4. The method `completeResults` will complete  futureA and futureB and futureC when read eventA. Howerver setB  has not been refresh to the policies cache. 
5. So setC maybe read the old B value

Reply:
Yes, this is the corner case for this solution, which can not handle with concurrent request for modifying topic policy. This solution fix the synchronize case. 
And i wonder whether we have the need to implement strong concurrent controll? There are two directions:
1. add a warning to users, that the topic policy setting api can not handle with concurrent cases
2. or, i can implement a stronger concurrent controll.
What do you think? @AnonHxy @Demogorgon314

Reply:
> This solution fix the synchronize case.

Maybe the synchronized case is not fixed yet :). 
For example, the result could still be unexpected  if we call the following three synchronized method. The `messageTTL` could be the old value after we updated `XXX`:
```
admin.topics().setRetention
admin.topics().setMessageTTL
admin.topics().setXXX
```